docs/queries/grouping.md

```diff
@@ -0,0 +1,97 @@
+---
+layout: default
+title: Grouping
+nav_order: 3
+parent: Queries
+---
+
+# Grouping
+{: .no_toc }
+
+<details open markdown="block">
+  <summary>
+    Table of contents
+  </summary>
+  {: .text-delta }
+1. TOC
+{:toc}
+</details>
+
+---
+
+## Basics
+
+By default, Tasks displays tasks in a single list.
+
+To divide the matching tasks up with headings, you can add `group by` lines to the query.
+
+### Available grouping properties
+
+You can group by the following properties:
+
+File locations:
+
+1. `path` (the path to the file that contains the task, that is, the folder and the filename)
+1. `folder` (the folder to the file that contains the task, which will be `/` for files in root of the vault)
+1. `filename` (the filename of the file that contains the task, without the `.md` extension)
+
+File contents:
+
+1. `linktext` (the text that would be shown in the task's backlink, but with no link added)
+1. `heading` (the heading preceding the task, or `(No heading)` if there are no headings in the file)
+
+Task properties:
+
+1. `status` (Done or Todo, which is capitalized for visibility in the headings)
+    * Note that the Done group is displayed before the Todo group,
+      which differs from the Sorting ordering of this property.
```

Comment:
Why not TODOs first?

Reply:
Odd.. I could have sworn I answered this before...

For now, there is no logic for sorting order of groups. They are purely sorted in alphabetical order by group name.

I think that adding the ability to sort groups logically would be good, but was feeling that this first implementation was useful enough to defer sorting (and reversing?) of groups until later...